java/flight/flight-jdbc-driver/src/main/java/org/apache/arrow/driver/jdbc/utils/UrlParser.java

```diff
@@ -37,13 +39,19 @@ public static Map<String, String> parse(String url, String separator) {
     String[] keyValues = url.split(separator);
 
     for (String keyValue : keyValues) {
-      int separatorKey = keyValue.indexOf("="); // Find the first equal sign to split key and value.
-      String key = keyValue.substring(0, separatorKey);
-      String value = "";
-      if (!keyValue.endsWith("=")) { // Avoid crashes for empty values.
-        value = keyValue.substring(separatorKey + 1);
+      try {
+        int separatorKey = keyValue.indexOf("="); // Find the first equal sign to split key and value.
+        String key = keyValue.substring(0, separatorKey);
+        key = URLDecoder.decode(key, "UTF-8");
```

Comment:
FYI, we ran into this problem and had a different fix for it in this PR which was to avoid using URLEncodedUtils.parse():
https://github.com/Bit-Quill/arrow/pull/3/commits/360ee52b6551fdc1c929f591ea1ae9e56e966333#diff-1884a04bc2b36e89427f1bbc51335c70d16d3f060e8af4019b168e5ab9458bdeL251

We wouldn't want users to have to manually URI encode key/values when building the connection string.

Reply:
Hmm, how do we want to manage this? Maybe merge the other PR into this one? Or vice-versa?

Reply:
Also, if the user isn't URI-encoding while building the string, how is the string parsed afterwards then? Wouldn't it be ambiguous?

Reply:
I'd be up for including the changes from  #3 into this PR.
Yeah, we haven't really tried URI-encoded key/values so this seems like it is necessary.

Reply:
Alright, I'll give it a shot…I'll probably undo the last commit first

Reply:
Ah, I see - since we're splitting it, it'll generally work/be tolerant of things so long as there's not a `&` or `%` in the parameters. I suppose that works. I'll keep the URL decode step and see about adding some tests.

Reply:
@jduo why were property names all lowercased? It's leading to test failures (e.g. because then the properties in the connection pool no longer compare equal to the properties in the connection itself)

Reply:
Ah, well, we can just insert some more lowercasing

Reply:
Names are lower-cased internally to make the properties case-insensitive in the Properties object and connection string.

Incidentally this adds some consistency with the ODBC driver because the C++ gRPC client throws an assertion if you try to transmit headers that are not lower-case.

Reply:
Got it. I just had to lowercase in the connection pool as well to make that consistent.